source/slang/check.cpp

```diff
@@ -3772,9 +3774,9 @@ namespace Slang
 
                     Name* tagAssociatedTypeName = getSession()->getNameObj("__Tag");
                     Decl* tagAssociatedTypeDecl = nullptr;
-                    if(auto enumTypeTypeDeclRefType = enumTypeType.As<DeclRefType>())
+                    if(auto enumTypeTypeDeclRefType = enumTypeType.dynamicCast<DeclRefType>())
```

Comment:
This one should probably be an `as`; it seems like an oversight that it wasn't doing a "canonical" cast in the original code.